perfkitbenchmarker/providers/gcp/util.py

```diff
@@ -268,6 +268,8 @@ def _AddCommonFlags(self, resource):
         self.flags['project'] = resource.project
       if hasattr(resource, 'zone') and resource.zone:
         self.flags['zone'] = resource.zone
+      if hasattr(resource, 'region') and resource.region:
```

Comment:
This could affect a lot of other benchmarks.  Recommend doing this as a smaller, reversible change, rather than part of this large one.

Reply:
This is something Chris suggested that we add. There was a lot of duplication of adding region to the commands for VPN. Adding this shouldn't be a problem, but we can add this as a smaller change later.

Reply:
It looks like the only other places in providers/gcp/* that set self.region also use that to set their cmd.flags. `grep -rn self.region perfkitbenchmarker/providers/gcp/` says gcp_dataproc.py, gcp_dpb_dataproc.py, gce_network.py, and gce_virtual_machine.py would be affected. 
I can move this commit [969602e ] to the top of the current PR and sqash everything else so it's easy to find, or roll back to previous state and make a new pr with this change and include the other affected files. Or something else?

Reply:
I'd prefer to do it as a separate PR.  Numerous parties have forks of the codebase, and if we're going to (usefully) change the contract on GcloudCommand, I'd rather do that as a standalone PR.  You could that that first if you want, but that would delay this PR.

Reply:
I pulled this commit out and rebased this PR.